src/Compilers/CSharp/Portable/FlowAnalysis/AbstractFlowPass.cs

```diff
@@ -1062,6 +1062,13 @@ public override BoundNode VisitGlobalStatementInitializer(BoundGlobalStatementIn
 
         public override BoundNode VisitLocalDeclaration(BoundLocalDeclaration node)
         {
+            if (node.DeclaredTypeOpt != null)
+            {
+                foreach (var dimension in node.DeclaredTypeOpt.BoundDimensionsOpt)
+                {
+                    VisitRvalue(dimension);
```

Comment:
>VisitRvalue(dimension); [](start = 20, length = 23)

We are not visiting ```ArgumentsOpt```, I don't think we should visit ```BoundDimensionsOpt``` either. These are misplaced expressions, they are not going to be executed.

Reply:
If we don't visit them, this has an unsigned variable error:

```csharp
int[M(out var x)] y;
var _ = X;

Reply:
> If we don't visit them, this has an unassigned variable error:

The call is never executed why would the variable be assigned? If you are concerned with the noise in baselines, just add a ```#pragma``` to suppress the undesired diagnostics.

---
In reply to: [260142577](https://github.com/dotnet/roslyn/pull/32843#discussion_r260142577) [](ancestors = 260142577)

Reply:
>  just add a ```#pragma``` to suppress the undesired diagnostics

This would work only for warnings, see Scope_DeclaratorArguments_01 in OutVarTests.cs for an example of an other way to filter out undesired diagnostics from base-lines.

---
In reply to: [260150627](https://github.com/dotnet/roslyn/pull/32843#discussion_r260150627) [](ancestors = 260150627,260142577)